tests/utilities/test_argparse_utils.py

```diff
@@ -48,3 +59,112 @@ def test_parse_args_from_docstring_empty():
         """
     )
     assert len(args_help.keys()) == 0
+
+
+def test_get_abbrev_qualified_cls_name():
+    assert get_abbrev_qualified_cls_name(Trainer) == "pl.Trainer"
+
+    class NestedClass:
+        pass
+
+    assert not __name__.startswith("pytorch_lightning.")
+    expected_name = f"{__name__}.test_get_abbrev_qualified_cls_name.<locals>.NestedClass"
+    assert get_abbrev_qualified_cls_name(NestedClass) == expected_name
+
+
+class AddArgparseArgsExampleClass:
+    """
+    Args:
+        my_parameter: A thing.
+    """
+
+    def __init__(self, my_parameter: int = 0):
+        pass
+
+    @staticmethod
+    def get_deprecated_arg_names() -> List[str]:
+        return []
+
+
+class AddArgparseArgsExampleClassViaInit:
+
+    def __init__(self, my_parameter: int = 0):
+        """
+        Args:
+            my_parameter: A thing.
+        """
+        pass
+
+
+class AddArgparseArgsExampleClassNoDoc:
+
+    def __init__(self, my_parameter: int = 0):
+        pass
+
+
+def extract_help_text(parser):
+    help_str_buffer = io.StringIO()
+    parser.print_help(file=help_str_buffer)
+    help_str_buffer.seek(0)
+    return help_str_buffer.read()
+
+
+@pytest.mark.parametrize(["cls", "name"], [
+    [AddArgparseArgsExampleClass, "AddArgparseArgsExampleClass"],
+    [AddArgparseArgsExampleClassViaInit, "AddArgparseArgsExampleClassViaInit"],
+    [AddArgparseArgsExampleClassNoDoc, "AddArgparseArgsExampleClassNoDoc"],
+])
+def test_add_argparse_args(cls, name):
+    """
+    Tests that ``add_argparse_args`` handles argument groups correctly, and
+    can be parsed.
+    """
+    parser = ArgumentParser()
+    parser_main = parser.add_argument_group("main")
+    parser_main.add_argument("--main_arg", type=str, default="")
+    parser_old = parser  # For testing.
+    parser = add_argparse_args(cls, parser)
+    assert parser is parser_old
+
+    # Check nominal argument groups.
+    help_text = extract_help_text(parser)
+    assert "main:" in help_text
+    assert "--main_arg" in help_text
+    assert f"{name}:" in help_text
+    assert "--my_parameter" in help_text
+    if cls is not AddArgparseArgsExampleClassNoDoc:
+        assert "A thing" in help_text
+
+    fake_argv = ["--main_arg=abc", "--my_parameter=2"]
+    args = parser.parse_args(fake_argv)
+    assert args.main_arg == "abc"
+    assert args.my_parameter == 2
+
+
+def test_negative_add_argparse_args():
+    with pytest.raises(RuntimeError, match="Please only pass an ArgumentParser instance."):
+        parser = ArgumentParser()
+        add_argparse_args(AddArgparseArgsExampleClass, parser.add_argument_group("bad workflow"))
+
+
+def test_add_argparse_args_no_argument_group():
+    """
+    Test similar to above, but with ``use_argument_group=False``
+    (old workflow).
```

Comment:
could you describe it without referring to "above"?

Reply:
Done in 285f2ec0